contracts/mocks/Disposables/DisposableCover.sol

```diff
@@ -35,15 +35,25 @@ contract DisposableCover is MasterAwareV2 {
   */
   uint32 public coverAssetsFallback;
 
-  function addProducts(Product[] calldata newProducts) public {
+  function addProducts(
+    Product[] calldata newProducts,
+    string[] calldata ipfsMetadata
+  ) external {
+    uint initialProuctsCount = products.length;
     for (uint i = 0; i < newProducts.length; i++) {
       products.push(newProducts[i]);
+      emit ProductUpserted(initialProuctsCount + i, ipfsMetadata[i]);
```

Comment:
spelling error here: initialProuctsCount -> initialProductsCount

Reply:
why is it upserted ?

You upsert if you may also update an existing record, but here it just pushes one at the end, so it would be just ProductAdded

Reply:
The event is emitted whenever a product changes. This means both adding a new product and editing an existing one. It's purpose is to follow changes to any specific wordings similar to what we currently have in the UI as "excluded scenarios" which vary from one product to another. This information is stored on IPFS and the event is a reference to the cid referencing the file that contains it.